src/canonicaljson/__init__.py

```diff
@@ -44,15 +44,17 @@ class Encoder(Protocol):  # pragma: no cover
     def encode(self, data: object) -> str:
         pass
 
-    def iterencode(self, data: object) -> Generator[str, None, None]:
+    def iterencode(self, data: object) -> Iterator[str]:
         pass
 
-    def __call__(self, *args: Any, **kwargs: Any) -> "Encoder":
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         pass
 
 
-class JsonLibrary(Protocol):
-    JSONEncoder: Encoder
+class JsonLibrary(Protocol):  # pragma: no cover
+    @property
+    def JSONEncoder(self) -> Type[Encoder]:
```

Comment:
Would `JSONEncoder: Type[Encoder]` have fixed it?

Reply:
Good question! Not AFAICS.

```
dmr on titan in python-canonicaljson on  dmr/typing-jsonmodule via 🐍 v3.11.1 (python-3.11.1) 
2023-02-15 20:17:14 ✔  $ git diff | cat
```

```diff
diff --git a/src/canonicaljson/__init__.py b/src/canonicaljson/__init__.py
index ef33332..6f27b99 100644
--- a/src/canonicaljson/__init__.py
+++ b/src/canonicaljson/__init__.py
@@ -52,9 +52,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
 
 class JsonLibrary(Protocol):  # pragma: no cover
-    @property
-    def JSONEncoder(self) -> Type[Encoder]:
-        pass
+    JSONEncoder: Type[Encoder]

 
```

```
dmr on titan in python-canonicaljson on  dmr/typing-jsonmodule via 🐍 v3.11.1 (python-3.11.1) 
2023-02-15 20:17:54 ✔  $ mypy
src/canonicaljson/__init__.py:146: error: Argument 1 to "set_json_library" has incompatible type Module; expected "JsonLibrary"  [arg-type]
src/canonicaljson/__init__.py:146: note: Following member(s) of "Module json" have conflicts:
src/canonicaljson/__init__.py:146: note:     JSONEncoder: expected "Type[Encoder]", got "Type[JSONEncoder]"
tests/test_canonicaljson.py:169: error: Argument 1 to "set_json_library" has incompatible type Module; expected "JsonLibrary"  [arg-type]
tests/test_canonicaljson.py:169: note: Following member(s) of "Module json" have conflicts:
tests/test_canonicaljson.py:169: note:     JSONEncoder: expected "Type[Encoder]", got "Type[JSONEncoder]"
Found 2 errors in 2 files (checked 3 source files)
```